experiments/AtmosLES/cfsite_hadgem2-a_07_amip.jl

```diff
@@ -0,0 +1,706 @@
+# Physics specific imports
+using ClimateMachine.Atmos: altitude, recover_thermo_state
+import ClimateMachine.Atmos: source!, atmos_source!, filter_source
+import ClimateMachine.BalanceLaws: source, eq_tends
+
+# Citation for problem setup
+## CMIP6 Test Dataset - cfsites
+## [Webb2017](@cite)
+
+"""
+    GCMRelaxation{PV, FT} <: TendencyDef{Source, PV}
+
+"""
+struct GCMRelaxation{PV <: Union{Mass, TotalMoisture}, FT} <:
+       TendencyDef{Source, PV}
+    τ_relax::FT
+end
+GCMRelaxation(::Type{FT}, args...) where {FT} = (
+    GCMRelaxation{Mass, FT}(args...),
+    GCMRelaxation{TotalMoisture, FT}(args...),
+)
+
+function source(
+    s::GCMRelaxation{Mass},
+    m,
+    state,
+    aux,
+    t,
+    ts,
+    direction,
+    diffusive,
+)
+    # TODO: write correct tendency
+    return 0
+end
+
+function source(
+    s::GCMRelaxation{TotalMoisture},
+    m,
+    state,
+    aux,
+    t,
+    ts,
+    direction,
+    diffusive,
+)
+    # TODO: write correct tendency
+    return 0
+end
+
+# Temperature tendency term, ∂T∂t
+
+"""
+    LargeScaleProcess{PV <: Union{Mass,Energy,TotalMoisture}} <: TendencyDef{Source, PV}
+
+# Energy tendency ∂_t ρe
+
+Temperature tendency for the LES configuration based on quantities
+from a GCM. Quantities are included in standard CMIP naming format.
+Tendencies included here are
+
+    tntha = temperature tendency due to horizontal advection
+    tntva = temperature tendency due to vertical advection
+    tntr = temperature tendency due to radiation fluxes
+    ∂T∂z = temperature vertical gradient from GCM values
+
+# Moisture tendency ∂_t ρq_tot
+
+Moisture tendency for the LES configuration based on quantities
+from a GCM. Quantities are included in standard CMIP naming format.
+Tendencies included here are
+
+    tnhusha = moisture tendency due to horizontal advection
+    tnhusva = moisture tendency due to vertical advection
+    ∂qt∂z = moisture vertical gradient from GCM values
+"""
+struct LargeScaleProcess{PV <: Union{Mass, Energy, TotalMoisture}} <:
+       TendencyDef{Source, PV} end
+
+LargeScaleProcess() = (
+    LargeScaleProcess{Mass}(),
+    LargeScaleProcess{Energy}(),
+    LargeScaleProcess{TotalMoisture}(),
+)
+
+function source(
+    s::LargeScaleProcess{Energy},
+    m,
+    state,
+    aux,
+    t,
+    ts,
+    direction,
+    diffusive,
+)
+    # Establish problem float-type
+    FT = eltype(state)
+    # Establish vertical orientation
+    k̂ = vertical_unit_vector(m, aux)
+    _e_int_v0 = e_int_v0(m.param_set)
+    # Unpack vertical gradients
+    ∂qt∂z = diffusive.lsforcing.∇ᵥhus
+    ∂T∂z = diffusive.lsforcing.∇ᵥta
+    w_s = aux.lsforcing.w_s
+    cvm = cv_m(ts)
+    # Compute tendency terms
+    # Temperature contribution
+    T_tendency = aux.lsforcing.Σtemp_tendency + ∂T∂z * w_s
+    # Moisture contribution
+    q_tot_tendency =
+        compute_q_tot_tend(m, state, aux, t, ts, direction, diffusive)
+
+    # TODO: Better names for these? Should we separate them?
```

Comment:
Can we address this TODO?

Reply:
I don't think we need to separate them. I will combine them together.